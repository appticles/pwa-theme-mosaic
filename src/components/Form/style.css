.ui.form > div {
  padding: 10px 0;
}

.ui.modal.transition.visible.active .content .ui.form > div input,
.ui.modal.transition.visible.active .content .ui.form > div textarea {
  margin: 5px 0 0 0;
}
