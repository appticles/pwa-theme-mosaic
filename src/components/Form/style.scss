@import "../../utilities/variables";

.comments-form.ui.form {
  > div {
    padding: $base-spacing-xsm 0;
    input,
    textarea {
      margin: $base-spacing-xs 0 0 0;
      border-color: $form-border-color;
    }
  }
}
