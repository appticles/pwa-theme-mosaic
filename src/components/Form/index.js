import React from 'react';
import { Field, reduxForm } from 'redux-form';
import { Form, Button } from 'semantic-ui-react';

import './style.css';

let CommentForm = (props) => {
  const { handleSubmit } = props;
  return (
    <Form onSubmit={handleSubmit}>
      <div>
        <label htmlFor="name">Name</label>
        <Field name="name" component="input" type="text" placeholder="Your name" required />
      </div>
      <div>
        <label htmlFor="email">Email</label>
        <Field name="email" component="input" type="email" placeholder="Your email address" required />
      </div>
      <div>
        <label htmlFor="content">Leave a comment</label>
        <Field name="content" component="textarea" type="text" placeholder="Tell us what you think..." required />
      </div>
      <Button type="submit" color="teal" compact>
        Submit
      </Button>
    </Form>
  );
};

CommentForm = reduxForm({
  form: 'comment',
})(CommentForm);

export default CommentForm;
