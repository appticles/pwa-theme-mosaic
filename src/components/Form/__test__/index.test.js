import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Form from '../index';

describe('Comment form', () => {
  it('should render correctly', () => {
    const mockHandleSubmit = jest.fn();
    const output = shallow(<Form onSubmit={mockHandleSubmit} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  // @todo Fix test
  /*
  it('submit works', () => {
    const testValues = {
      author_name: 'Name',
      email: 'email@mail',
      content: {
        rendered: 'Comment content',
      },
      date: new Date(),
      post: '53',
      handleSubmit: jest.fn(),
    };
    const component = shallow(<Form {...testValues} />);
    component.debug();
    component.find('#submit').simulate('click');
    expect(testValues.handleSubmit).toHaveBeenCalledTimes(1);
    expect(testValues.handleSubmit).toBeCalledWith({
      author_name: testValues.author_name,
      email: testValues.email,
      content: {
        rendered: testValues.content.rendered,
      },
      date: testValues.date,
      post: testValues.post,
      handleSubmit: testValues.handleSubmit,
    });
  });
  */
});
