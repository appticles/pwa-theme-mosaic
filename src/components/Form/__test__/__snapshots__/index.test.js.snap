// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment form should render correctly 1`] = `
<Connect(Form(CommentForm))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="comment"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  onSubmit={[Function]}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
